pkgs/development/python-modules/vector/default.nix

```diff
@@ -1,58 +1,77 @@
 {
   lib,
-  awkward,
   buildPythonPackage,
-  fetchPypi,
+  pythonOlder,
+  fetchFromGitHub,
+
+  # build-system
   hatch-vcs,
   hatchling,
-  numba,
+
+  # dependencies
   numpy,
-  notebook,
   packaging,
+
+  # checks
+  awkward,
+  dask-awkward,
+  notebook,
+  numba,
   papermill,
   pytestCheckHook,
-  pythonOlder,
+  sympy,
+
+  stdenv,
 }:
 
 buildPythonPackage rec {
   pname = "vector";
   version = "1.4.1";
-  format = "pyproject";
+  pyproject = true;
 
   disabled = pythonOlder "3.8";
 
-  src = fetchPypi {
-    inherit pname version;
-    hash = "sha256-Fa74kRVg2x6j/6nb1UFNDsV1pQSiw/I+pFFwoYmURm4=";
+  src = fetchFromGitHub {
+    owner = "scikit-hep";
+    repo = "vector";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-ckwJgh5aWFTwoEBVmKukdk+cmQeOuecLiYD2EcuELGw=";
   };
 
-  nativeBuildInputs = [
+  build-system = [
     hatch-vcs
     hatchling
   ];
 
-  propagatedBuildInputs = [
+  dependencies = [
     numpy
     packaging
   ];
 
-  checkInputs = [
+  nativeCheckInputs = [
     awkward
+    dask-awkward
     notebook
```

Comment:
Doesn't seem to depend on notebook.

Reply:
I agree, it doesn't appear on the [pyproject.toml](https://github.com/scikit-hep/vector/blob/v1.4.1/pyproject.toml).
However, a test fails without it:
https://pastebin.com/fv4DF5Sy